internal/distro/rhel9/imagetype.go

```diff
@@ -314,32 +316,34 @@ func (t *imageType) PackageSets(bp blueprint.Blueprint, options distro.ImageOpti
 }
 
 // checkOptions checks the validity and compatibility of options and customizations for the image type.
-func (t *imageType) checkOptions(customizations *blueprint.Customizations, options distro.ImageOptions, containers []container.Spec) error {
+func (t *imageType) checkOptions(customizations *blueprint.Customizations, options distro.ImageOptions, containers []container.Spec) ([]string, error) {
 
+	// holds warnings (e.g. deprecation notices)
+	warnings := []string{}
```

Comment:
I suggest to define `warnings` as
```go
var warnings []string
```
and then just `append()` to it. This way, the function will return `nil` if there are not warnings, instead of an empty slice. So the behavior is now inconsistent with the same same functions defined in distros that do not return any warnings.

Reply:
Right, so if there are no warnings, just return nil; but, if there is an error, even if we have warnings, just return nil since the error takes priority?

Reply:
Yes, I should have been consistent between fedora+rhel7/8/9, since we want the same thing; so changing it to `var warnings []string` ensures that `nil` when there are no warnings (as in fedora + rhel7) and it initialises the slice with whatever warning there is due to the `append` when we are on rhel8/9; we agree there :+1: